### PR TITLE
v5/docs: reintroduce `outline` for docs code samples, buttons when `:not(:focus-visible)`

### LINKED ISSUE #36506: v5 docs: missing (clear) visible focus indication for version switch button and code samples
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

In the current https://getbootstrap.com/ site, the version switch button in the header has a minimal focus indication (just the text colour getting a bit lighter).

Further down in the page, any `<pre>` code sample is focusable, but focus indication has been suppressed.

https://user-images.githubusercontent.com/895831/172058438-ee48720d-b080-4dce-b671-f8ab70ad288a.mp4

### Reduced test cases

https://getbootstrap.com/

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

5.2

### PR DESCRIPTION
**Pull request by patrickhlauke**
closes https://github.com/twbs/bootstrap/issues/36506

### CODE REVIEW

**Comment by patrickhlauke** [comment]
https://user-images.githubusercontent.com/895831/172059262-0d06623d-0523-4e14-9a45-9a07b25cdd00.mp4

**Comment by patrickhlauke** [comment]
@mdo et al, any thoughts on this?

File: scss/_buttons.scss [hunk 1]
```diff
@@ -60,6 +59,11 @@
     }
   }
 
+  .btn-check:focus:not(:focus-visible) + &,
+  &:focus:not(:focus-visible) {
+    outline: 0;
```
Comment (ffoodd) [inline scss/_buttons.scss:64]: Shouldn't we go with `outline-color: transparent` instead, to improve high contrasts mode support ?

Comment (patrickhlauke) [reply scss/_buttons.scss:64]: oh, do we do this elsewhere already? otherwise, i'd say let's do it consistently, but in a separate PR

Comment (patrickhlauke) [reply scss/_buttons.scss:64]: answering my own question: no, seems we're using `outline: 0` everywhere at the moment. so yes, I'd say let's park that thought, and then do it consistently throughout in a new PR specifically for WHCM improvements

**Comment by patrickhlauke** [comment]
any chance we could get this looked at @twbs/css-review @mdo ?

**Comment by julien-deramond** [comment]
Focus indication on `<pre>` code samples is now visible. Great!

Screenshot from 2022-07-24 08-48-26

**Note:** We should improve the rendering because the focus area is under the copy button

---

Regarding the buttons, the rendering on Chrome is good. On Firefox, however, the box shadow + red outline combination hurts the eyes a little bit. Maybe just because I'm not used to it yet.

Screenshot from 2022-07-24 08-56-43
Screenshot from 2022-07-24 09-16-25

Screenshot from 2022-07-24 08-58-27
Screenshot from 2022-07-24 08-59-12

Screenshot from 2022-07-24 09-00-12

---

**Off-topic**: while playing with the documentation, focus indication for carousels controls; not visible enough IMO.

**Comment by patrickhlauke** [comment]
in Chrome, there seems to be a tiny bit of overlap as well ... i can have a look at nudging something by a few pixels to compensate.

image

in my Firefox/Win, it uses blue rather than red for some reason. maybe we should explicitly define the outline colour as well at some point (part of the larger reevaluation of focus/focus-visible styles perhaps)

image

**Comment by julien-deramond** [comment]
Let's do it step by step, no problem. In other PRs:
* nudging something by a few pixels to compensate (if too complicated for this PR)
* defining the outline colour to be coherent between browsers (part of the larger reevaluation of focus/focus-visible styles)

**Review by julien-deramond — APPROVED** [review]
LGTM! @mdo I let you check this as well.

**Comment by patrickhlauke** [comment]
I'll have a look at the subtle nudging now as part of this PR, so hold off merging until I had a crack at it

**Comment by patrickhlauke** [comment]
Done...added an extra bit of explicit margin on the right-hand side of the `pre`.

In Firefox now:

image

In Chrome:

image

As said, leaving the outline color thing for another time (as it will likely need to be done on a section basis, so the outline will be a light colour in the header, and a dark one in the main section, etc)

So this PR is now ready for final review/merging

**Comment by patrickhlauke** [comment]
@mdo et al, any news on this?

**Review by mdo — APPROVED** [review]

**Pull request merged**